core/src/test/java/kafka/admin/AdminFenceProducersTest.java

```diff
@@ -97,9 +97,11 @@ void testFenceAfterProducerCommit() throws Exception {
                     ExecutionException.class,
                     () -> producer.send(RECORD).get(), "expected ProducerFencedException"
             );
-            assertInstanceOf(ProducerFencedException.class, exceptionDuringSend.getCause());
```

Comment:
Is the comment in line 98 correct? And did we change the error that returns here?

Reply:
Is it because we hit the epoch check in the produce path before the add partitions path?